src/qiboml/backends/pytorch.py

```diff
@@ -15,20 +15,28 @@ class TorchMatrices(NumpyMatrices):
         dtype (torch.dtype): Data type of the matrices.
     """
 
-    def __init__(self, dtype):
+    def __init__(self, dtype, device):
```

Comment:
Maybe it is worth defaulting `device` to `None` and globally setting it to either GPU or CPU using https://pytorch.org/docs/stable/generated/torch.set_default_device.html